homeassistant/components/matter/switch.py

```diff
@@ -80,4 +105,24 @@ def _update_from_device(self) -> None:
             device_types.Thermostat,
         ),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.SWITCH,
+        entity_description=SwitchEntityDescription(
+            key="MatterGenericSwitch", device_class=SwitchDeviceClass.SWITCH, name=None
+        ),
+        entity_class=MatterGenericSwitch,
+        required_attributes=(clusters.Switch.Attributes.CurrentPosition,),
+        # restrict device type to prevent discovery by the wrong platform
+        not_device_type=(
+            device_types.ColorTemperatureLight,
+            device_types.DimmableLight,
+            device_types.ExtendedColorLight,
+            device_types.OnOffLight,
+            device_types.DoorLock,
+            device_types.ColorDimmerSwitch,
+            device_types.DimmerSwitch,
+            device_types.OnOffLightSwitch,
+            device_types.Thermostat,
+        ),
+    ),
```

Comment:
Because the required attribute is CurrentPosition and the deviceclass explictly set, you may drop these

Reply:
Done